docs/DeveloperGuide.adoc

```diff
@@ -314,16 +370,144 @@ _{More to be added}_
+[discrete]
+=== Use case: Add person
+Guarantees: +
+
+1. Person is added even if input fields might be invalid (see 1a).
+
+*MSS*
+
+1.  User requests to add a person.
+2.  AddressBook adds the person.
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. Either of the given NRIC, contact number, or email address is invalid.
++
+[none]
+** 1a1. AddressBook adds the person into address book.
+** 1a1. AddressBook shows a warning.
+
+* 1b. Duplicate profile is added.
++
+[none]
+** 1b1. AddressBook shows an error message and will attempt to merge the profile.
++
+
 
 [discrete]
-=== Use case: Delete person
+=== Use case: Edit person
+Guarantees: +
+
+1. Person is editted even if input fields might be invalid (see 3a).
+
+*MSS*
+
+1.  User requests to find a person
+2.  AddressBook shows details of the person
+3.  User requests to edit that specific person.
+4.  AddressBook edits the person.
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. Person is not found
++
+[none]
+** 1a1. AddressBook shows an error message.
++
+Use case resumes at step 1.
+
+* 3a. Either of the given NRIC, contact number, or email address is invalid.
++
+[none]
+** 3a1. AddressBook adds the person into address book.
+** 3a2. AddressBook shows a warning.
++
+
+* *a. At any time, User requests to cancel the edit.
++
+[none]
+** *a1. Address books requests to confirm the cancellation.
+** *a1. User confirms the cancellation.
+Use case ends.
+
+[discrete]
+=== Use case: Delete contact
+
+*MSS*
+
+1.  User requests to list contacts
+2.  AddressBook shows a list of contacts
+3.  User requests to delete a specific contact in the list
+4.  AddressBook deletes the contact
+5.  Contact appears in the recycling bin
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. The list is empty.
++
+Use case ends.
+
+* 3a. The given index is invalid.
++
+[none]
+** 3a1. AddressBook shows an error message.
++
+Use case resumes at step 2.
+
+[discrete]
+=== Use case: Assigning a policy to a person
 
 *MSS*
 
 1.  User requests to list persons
 2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+3.  User requests to assign a policy to a specific person
+4.  The policy gets assigned to the person
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. The list is empty.
++
+Use case ends.
+
+* 3a. The given index is invalid.
++
+[none]
+** 3a1. AddressBook shows an error message.
++
+Use case resumes at step 2.
+
+* 3b. The person is not eligible for the policy.
++
+[none]
+** 3b1. AddressBook shows an error message and returns a list of policies the person is eligible for.
++
+Use case resumes at step 2.
+
+
+[discrete]
+=== Use case: Restoring recently deleted contact
+
+*MSS*
+
+1.  User requests to list recently deleted contacts from recycling bin
+2.  AddressBook shows a list of contacts
+3.  User requests to restore a specific contact in the list
```

Comment:
Maybe you could include an extension to cover the case where there is an error in specifying the contact in the list (e.g. wrong identifier for the contact because of a typo)?